src/components/ec-text-field/ec-text-field.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="ec-text-field">
+    <div
+      v-if="label"
+      class="ec-text-field__label"
+    >{{ label }}</div>
+    <input
+      class="ec-text-field__input"
+      :class="{
+        'ec-text-field__input--error': state === 'error',
```

Comment:
DRY. `state === 'error'` is used multiple times in the template, move it to computed property, don't do logic in the template.

Reply:
+1, also add a hash map to the top with the different statuses, and use it such `STATUSES.ERROR`

maybe in the future we could have a HOC (Higher-order component) for components with a definite set of statuses